src/git-version.ts

```diff
@@ -0,0 +1,63 @@
+export class GitVersion {
+  private readonly major: number = NaN
+  private readonly minor: number = NaN
+  private readonly patch: number = NaN
+
+  constructor(version?: string) {
+    if (version) {
+      const match = version.match(/^(\d+)\.(\d+)(\.(\d+))?$/)
+      if (match) {
+        this.major = Number(match[1])
+        this.minor = Number(match[2])
+        if (match[4]) {
+          this.patch = Number(match[4])
+        }
+      }
+    }
+  }
+
+  checkMinimum(minimum: GitVersion): boolean {
```

Comment:
I would add access modifiers to these methods to make them clear

Reply:
+1 however the prettier config didnt like it.